extmod/uasyncio/stream.py

```diff
@@ -40,7 +40,15 @@ async def readline(self):
                 return l
 
     def write(self, buf):
-        self.out_buf += buf
```

Comment:
FYI: this will not allocate memory if `out_buf` is `b""`, the runtime is optimised for that case

Reply:
Thanks for pointing that out, I noticed it after I started hacking around and then continued because I looked for the semantics of `write` and found https://docs.python.org/3/library/io.html#io.RawIOBase.write which states "The caller may release or mutate buf after this method returns, so the implementation should only access buf during the method call." The docs for `write` in asyncio are thin, so there may be some room for interpretation about whether it must adhere to the above semantics.